ksql-serde/src/main/java/io/confluent/ksql/serde/connect/ConnectDataTranslator.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.serde.connect;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.util.KsqlException;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ConnectDataTranslator {
+  public GenericRow toKsqlRow(final Schema schema, final Schema connectSchema,
+                              final Object connectData) {
+    if (! schema.type().equals(Schema.Type.STRUCT)) {
+      throw new KsqlException("Schema for a KSQL row should be a struct");
+    }
+    final Struct rowStruct = (Struct) toKsqlValue(schema, connectSchema, connectData);
```

Comment:
should check for null here and throw an appropriate exception

Reply:
Its actually valid for rowStruct to be null if the row itself is null. I'll change this to return null in that case.